src/chk/chk_vos.c

```diff
@@ -78,17 +78,7 @@ chk_db_delete(char *key, int key_size)
 static int
 chk_db_traverse(sys_db_trav_cb_t cb, void *args)
 {
-	int	rc;
-
-	if (chk_db->sd_lock)
-		chk_db->sd_lock(chk_db);
```

Comment:
This locking is necessary, it can't be simply removed here. ABT_mutex does support nested locking, so it's ok even if the traverse callback lock it again.
The deadlock can only happen when traverse callback create a new ULT and take the same lock in the new ULT, we should avoid such use case.

Reply:
It is hope that the `sysdb` lock can be held recursively, but the current implementation does not. I have refreshed the patch to change the sysdb lock attribute.